patroni/postgresql/__init__.py

```diff
@@ -984,17 +989,22 @@ def pick_synchronous_standby(self, cluster, sync_node_count=1):
         # receiving changes faster than the sync member (very rare but possible). Such cases would
         # trigger sync standby member swapping frequently and the sort on sync_state desc should
         # help in keeping the query result consistent.
-        for app_name, state, sync_state in self.query(
-                "SELECT pg_catalog.lower(application_name), state, sync_state"
+        for app_name, state, sync_state, replica_lsn in self.query(
+                "SELECT pg_catalog.lower(application_name), state, sync_state, pg_{2}_{1}_diff({0}_{1}, '0/0')::bigint"
                 " FROM pg_catalog.pg_stat_replication"
                 " WHERE state = 'streaming'"
-                " ORDER BY sync_state DESC, {0}_{1} DESC".format(sort_col, self.lsn_name)):
+                " ORDER BY sync_state DESC, {0}_{1} DESC".format(sort_col, self.lsn_name, self.wal_name)):
             member = members.get(app_name)
-            if not member or member.tags.get('nosync', False):
-                continue
-            candidates.append(member.name)
-            if sync_state == 'sync':
-                sync_nodes.append(member.name)
+            if member and not member.tags.get('nosync', False):
+                replica_list.append((member.name, state, sync_state, replica_lsn))
```

Comment:
Since the `state` is not used anymore outside the query I think it could be removed.